### ISSUE TITLE
I original file extension, which i have uploaded

### ISSUE THREAD

**Issue by RonakShishangia** [tus/tusd#174]
when upload any file it will upload with ".bin" file but i want that original file name. Please give me your valuable response.

**Comment by Acconut** [comment]
You can attach the original file name as metadata to the upload and use the `post-finish` hook (see https://github.com/tus/tusd/blob/master/docs/hooks.md#post-finish) to extract the filename and rename the `.bin` file.

**Comment by zhangalex** [comment]
A little complicated.

**Comment by Acconut** [comment]
Yes, I kind of agree. Do you have a better idea? If so, I am always happy to hear them

**Comment by zhangalex** [comment]
Why not allow developer custom filename rule while uploading? For example,

`
uploader.filename(e->{
  return UUID.randomString() + "." + e.getExtension();
});
`
If set up the filename rule interface, it will automatically override the default filename rule. So, the developer will guarantee the unique of filename.

**Comment by Acconut** [comment]
I think your approach is very similar to renaming the file inside a hook. The only difference is where the code for the name generation is located. With your way, it's done inside tusd directly. With the other way, it's located inside a different executable. So, frankly, I do not see much benefit there.

**Comment by zhangalex** [comment]
The benefit is that it will save a lot of time.

**Comment by Acconut** [comment]
Sometimes it's not worth to sacrifice simplicity and flexibility for small time benefits, IMO. However, we are always happy to accept PRs for improvements.

**Comment by iuriikomarov** [comment]
+1 to the feature. Would love to see ability to compute object names within `store` or `composer` instance.

**Comment by Acconut** [comment]
Thanks for the feedback. We have it on a wishlist but since we are quite tight in time, I doubt we will come to it soon.
